content/blog/getting-started-with-k8s-part6/index.md

```diff
@@ -0,0 +1,53 @@
+
+In addition to zero downtime cluster migrations, we can also control scaling. The following tutorial shows how to stage an application rollout from dev to testing controlled by performance collecting metrics. In this [tutorial]({{< relref "/docs/tutorials/kubernetes/p8s-rollout" >}}), we move a three replica canary to a ten replica staging deployment. Replicas are added based on a P90 response time, i.e., 90% of requests are processed within 2.75 seconds. The example uses [Prometheus](https://prometheus.io/) to collect the response times and a [utility class](https://github.com/pulumi/examples/blob/master/kubernetes-ts-staged-rollout-with-prometheus/util.ts) to perform the health check. You can clone the [code on Github](https://github.com/pulumi/examples/tree/master/kubernetes-ts-staged-rollout-with-prometheus).
+
+Pulumi also lets you update your Kubernetes resources in real-time with Pulumi watch. You can add more replicas to deployment by saving changes to the stack. Adding a Service to make an application available outside the cluster with a load balancer is instantaneous. Watch how you can update your application.
+
+{{< youtube "X96EMLi8uJY?rel=0" >}}
+
+## Monitoring
+
+There are many [logging and monitoring](https://kubernetes.io/docs/tasks/debug-application-cluster/) solutions available for Kubernetes. In the previous section, we used Prometheus for generating metrics. Pulumi supports logging and monitoring tools to assist with cluster and application management.
+
+Kubernetes Cluster Services provide logging and monitoring at the cluster level or a subset of apps and workloads. Tutorials for setting up logging and monitoring for Kubernetes on [AWS, Azure, and GCP]({{< relref "/docs/guides/crosswalk/kubernetes/cluster-services" >}}). The tutorial also shows how to configure [DataDog](https://www.datadoghq.com/) to log and monitor applications. The [example code](https://github.com/pulumi/kubernetes-guides/tree/master/general-cluster-services/datadog-daemonset) is on Github.
+
+Pulumi also includes tools for managing resources. [Kubernetes Query](https://github.com/pulumi/pulumi-query-kubernetes) supports both batch and streaming queries that lets you ask questions such as, “How many Pods are in a Node?” Batch queries run to completion and return results based on the current configuration. Streaming queries continuously watch the cluster and take action based on criteria; for example, a streaming query can watch storage and warn you if the storage is 80% full. You can read about typical use cases on the Pulumi [blog]({{< relref "/blog/query-kubernetes" >}}).
```

Comment:
Do we want to mention kquery? It's not actively maintained, and I don't believe it really fits here. Do you believe it fits inline with the day2 story?

Reply:
I agree it should be cut if we're not maintaining it.